src/legacy/ui/ui_settings/ui_settings_mixin.js

```diff
@@ -27,9 +27,22 @@ import {
 } from './routes';
 
 export function uiSettingsMixin(kbnServer, server) {
-  const getDefaults = () => (
-    kbnServer.uiExports.uiSettingDefaults
-  );
+  const getDefaults = () => {
+    return Object.keys(kbnServer.uiExports.uiSettingDefaults).reduce((acc, currentKey) => {
+      const defaultSetting = kbnServer.uiExports.uiSettingDefaults[currentKey];
+      const updatedDefaultSetting = {
+        ...defaultSetting,
+      };
+      if (typeof defaultSetting.options === 'function') {
+        updatedDefaultSetting.options = defaultSetting.options(server);
```

Comment:
Since this value only has access to the value of server, I think it makes sense to merge the defaults once, rather than every time the defaults are requested.

Reply:
Yes that was also my feelings, but I wasn't sure if the `kbnServer.uiExports.uiSettingDefaults` object can be considered final at the beginning of that `uiSettingsMixin` function.
If so I can apply that merge at the beginning of the function and use that merged object inside the getDefaults function. Is that what you mean @spalger ?

Reply:
Yep! `kbnServer.uiExports` is created and never modified, the data necessary to populate it is available once the plugins have been discovered, which is one of the first things the server does.

Reply:
@spalger From the CI failures seems that `kbnServer.uiExports.uiSettingDefault` is undefined. I've updated the PR with a default `{}` object, but I want to check with you if that case can really exist or if we miss something in our tests.

Reply:
That should only be possible if there aren't any `uiSettingDefault` uiExports, which will only happen in the tests because we disable all plugins sometimes. Defaulting to an empty object is the right approach here 👍